File: Agentic-AI-Travel-Advisor/shared/Core/Entities/TravelPreferences.cs
namespace TravelAdvisor.Core.Entities;

public class TravelPreferences
{
    public int Id { get; set; }
    public string UserId { get; set; } = string.Empty;
    public decimal? BudgetMin { get; set; }
    public decimal? BudgetMax { get; set; }
    public string? PreferredClimate { get; set; }
    public string? Interests { get; set; }

    public ApplicationUser User { get; set; } = null!;
}

File: Agentic-AI-Travel-Advisor/mobile-app/lib/models/travel_preferences.dart
class TravelPreferences {
  final double? budgetMin;
  final double? budgetMax;
  final String? preferredClimate;
  final String? interests;

  const TravelPreferences({
    this.budgetMin,
    this.budgetMax,
    this.preferredClimate,
    this.interests,
  });

  factory TravelPreferences.fromJson(Map<String, dynamic> json) {
    return TravelPreferences(
      budgetMin: (json['budgetMin'] as num?)?.toDouble(),
      budgetMax: (json['budgetMax'] as num?)?.toDouble(),
      preferredClimate: json['preferredClimate'] as String?,
      interests: json['interests'] as String?,
    );
  }
}
